src/main/java/ch/njol/skript/expressions/ExprBossBar.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * This file is part of Skript.
+ * <p>
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * <p>
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * <p>
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ * <p>
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.skript.util.Color;
+import ch.njol.util.Kleenean;
+import org.bukkit.Bukkit;
+import org.bukkit.DyeColor;
+import org.bukkit.boss.BarColor;
+import org.bukkit.boss.BarStyle;
+import org.bukkit.boss.BossBar;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Boss Bar")
+@Description("")
+@Examples({
+	"set {bar} to a new boss bar",
+	"set the name of {bar} to \"hello\"",
+	"set the color of {bar} to red",
+	"add player to {bar}",
+	"set {bar} to a new pink boss bar named \"hello\""
+})
+@Since("INSERT VERSION")
+public class ExprBossBar extends SimpleExpression<BossBar> {
+
+	static {
+		Skript.registerExpression(ExprBossBar.class, BossBar.class, ExpressionType.SIMPLE,
+			"[a] new boss[ ]bar [name:(named|titled|with title) %-string%]",
+			"[a] new %color% boss[ ]bar [name:(named|titled|with title) %-string%]"
```

Comment:
```suggestion
			"[a] new boss[ ]bar [(named|titled|with title) %-string%]",
			"[a] new %color% boss[ ]bar [(named|titled|with title) %-string%]"
```
You don't need the parse tags, and you can remove the check from init.
Also what do you think about supporting `with name` as an alternative?